hivemq4/dns-image/Dockerfile

```diff
@@ -14,14 +14,17 @@ ENV HIVEMQ_CLUSTER_TRANSPORT_TYPE UDP
 
 COPY config-dns.xml /opt/hivemq/conf/config.xml
 
-RUN mkdir /opt/hivemq/extensions/hivemq-dns-cluster-discovery
-COPY hivemq-dns-cluster-discovery/* /opt/hivemq/extensions/hivemq-dns-cluster-discovery/
+RUN curl -L https://github.com/hivemq/hivemq-dns-cluster-discovery-extension/releases/download/4.1.0/hivemq-dns-cluster-discovery-4.1.0.zip -o /opt/hivemq/extensions/dns-discovery.zip \
+    && unzip /opt/hivemq/extensions/dns-discovery.zip -d /opt/hivemq/extensions \
+    && rm -f /opt/hivemq/extensions/hivemq-dns-cluster-discovery/*.png \
+    && chgrp -R 0 /opt/hivemq/extensions/hivemq-dns-cluster-discovery \
+    && chmod -R 770 /opt/hivemq/extensions/hivemq-dns-cluster-discovery \
+    && rm /opt/hivemq/extensions/dns-discovery.zip
 
 COPY pre-entry.sh /opt/pre-entry.sh
 
 RUN chmod +x /opt/docker-entrypoint.sh \
-    && chmod +x /opt/pre-entry.sh \
-    && chown -R hivemq:hivemq /opt/hivemq
```

Comment:
Why remove the ownership here ?

Reply:
we shouldn't change ownership at build-time because it decreases layer efficency and the UID of the hivemq user within the container is pretty much never the UID at runtime.
Also openshift container guidelines state that
> For an image to support running as an arbitrary user, directories and files that are written to by processes in the image must be owned by the root group and be read/writable by that group. Files to be executed must also have group execute permissions.
https://docs.openshift.com/container-platform/4.7/openshift_images/create-images.html#images-create-guide-openshift_create-images

Reply:
@schaebo  - is this clarified on your side or do you need further input?